wayland/globals.c

```diff
@@ -146,6 +149,14 @@ static void awesome_handle_global(void *data, struct wl_registry *registry,
         globalconf.wl_shm = wl_registry_bind(registry, name,
                 &wl_shm_interface, version);
     }
+    else if (strcmp(interface, wl_output_interface.name) == 0)
+    {
+        struct wl_output *wl_output =
+            wl_registry_bind(registry, name, &wl_output_interface, version);
+
+        lua_State *L = globalconf_get_lua_State();
+        screen_add(L, &globalconf.screens, wl_output);
```

Comment:
(copy pasting the comment I already added in the commit)

Do you think you could rebase this on top of my screen branch and change this to create outputs instead of screens? That branch is quite far in the reviews, so I don't think it will change much. This will be very hard to rebase (along with the x11/screen.c when time comes to merge both changeset. Having the information as outputs and not screen is very important to support HiDPI later on.

I propose you remove the --screen on support in the wayland version and always use --screen off and use the viewports all the time. The extra abstraction is required to better support plugging and unplugging screen very often (work laptops attached to larger screen with multiple meeting per days and/or student making presentation on projectors).